src/base.rs

```diff
@@ -115,15 +121,39 @@ pub struct Base32UpperPad;
 
 impl BaseImpl for Base32UpperPad {
     fn encode(input: &[u8]) -> String {
-        base32::encode(base32::Alphabet::RFC4648 { padding: true }, input)
+        BASE32.encode(input)
     }
 
     fn decode(input: &str) -> Result<Vec<u8>> {
-        if let Some(result) = base32::decode(base32::Alphabet::RFC4648 { padding: true }, input) {
-            Ok(result)
-        } else {
-            Err(Error::InvalidBaseString)
-        }
+        Ok(BASE32.decode(input.as_bytes())?)
```

Comment:
You don't need to wrap in in an Ok. This should also work: `BASE32.decode(input.as_bytes())` as it also returns a result.

There are more occurrences of this further down, I won't mention it there again.

Reply:
So the thing is that this `Result` wants the `multibase::Error` defined in `error.rs:6`and the only other way I know how to make the compiler happy is by writing:

`BASE32.decode(input.as_bytes()).map_err(Into::into)`

Don't know if you have a preference there?

Reply:
Took a look at the generated code, and it looks like the `Ok(BASE32.decode(input.as_bytes())?)` is optimized better for some reason. I'll leave it as it is.

Reply:
I thought implicit conversion would happen here. I also don't see much difference in the compiled code, hence I would say it doesn't really matter and I also don't have a preference.